.github/workflows/wasmcloud-chart.yml

```diff
@@ -0,0 +1,83 @@
+name: wasmcloud-chart
+
+env:
+  HELM_VERSION: v3.13.1
+
+on:
+  push:
+    tags:
+      - 'chart-v[0-9].[0-9]+.[0-9]+'
+  pull_request:
+    paths:
+      - 'chart/**'
+      - '.github/workflows/wasmcloud-chart.yml'
+
+jobs:
+  validate:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: actions/checkout@v4.1.1
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v3.5
+        with:
+          version: ${{ env.HELM_VERSION }}
+
+      # Used by helm chart-testing below
+      - name: Set up Python
+        uses: actions/setup-python@v4.7.1
+        with:
+          python-version: '3.12.0'
+
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v2.4.0
+        with:
+          version: v3.9.0
+          yamllint_version: 1.32.0
+          yamale_version: 4.0.4
+
+      - name: Run chart-testing (lint)
+        run: |
+          ct lint --charts chart/ --validate-maintainers=false
+
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.8.0
+
+      - name: Run chart-testing (install)
+        run: ct install --charts chart/
+
+  publish:
+    if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/chart-v') }}
+    runs-on: ubuntu-22.04
+    needs: validate
+    permissions:
+      packages: write
+
+    steps:
+      - uses: actions/checkout@v4.1.1
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v3.5
+        with:
+          version: ${{ env.HELM_VERSION }}
+
+      - name: Package
+        run: |
+          helm package chart -d .helm-charts
+
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish
+        run: |
+          for chart in .helm-charts/*; do
+            if [ -z "${chart:-}" ]; then
+              break
+            fi
+            helm push "${chart}" "oci://ghcr.io/${{ github.repository_owner }}"
```

Comment:
I don't have any experience publishing charts, but did notice for the [old chart](https://github.com/wasmCloud/wasmcloud-otp/blob/main/.github/workflows/release.yml) we use the `helm/chart-releaser-action` action. Is there a reason we shouldn't use that anymore?

Reply:
So the [`helm/chart-releaser-action`](https://github.com/helm/chart-releaser-action) is primarily intended for publishing charts to be self-hosted on GitHub Pages, which is not how we intend to host the chart. Instead, we're looking to publish the chart onto GitHub Container Registry as an OCI artifact.

The `chart-releaser-action` has an [issue open regarding adding support for OCI registries](https://github.com/helm/chart-releaser-action/issues/107), which would let us use it in the future should they choose to implement that functionality.

Reply:
Ah, got it! Thanks 👍